### PR TITLE
tests: new remote tools - part 1

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change introduces the new tools for remote commands from the snapd-testing-tools project

This change starts using the remote.exec which replaces the tests.nested exec and the remote.push by the tests.nested copy

The new remote commands have the same implementation then the tests.nested but those include tests and are being used in other projects successfully.

### CODE REVIEW

File: tests/lib/external/snapd-testing-tools/remote/remote.refresh [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    check_ubuntu_core
+    snap_name="$(remote.exec "snap list" | grep -E '(kernel$|kernel,)' | awk '{ print $1 }')"
+    if [ -z "$snap_name" ]; then
+        echo "remote.refresh: no kernel snap to update"
+        return
+    fi
+
+    process_refresh "$snap_name" "$refresh_channel"
+}
```
Comment (mardy) [inline tests/lib/external/snapd-testing-tools/remote/remote.refresh:73]: There's a lot of common code between this function and the ones below. I think we should try to make them call a common helper. Like this:
```suggestion
refresh_kernel() {
    check_ubuntu_core
    refresh_snap "$1"
}
```

File: tests/lib/external/snapd-testing-tools/tools/tests.systemd [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    if [ -z "$start_line" ]; then
+        echo "tests.systemd: unit command line cannot be empty"
+        return 1
+    fi
+    if [ -f "/etc/systemd/system/$name.service" ]; then
+        echo "tests.systemd: unit service file already exist"
+        return 1  
+    fi
+
+    printf '[Unit]\nDescription=Support for test %s\n[Service]\nType=simple\nExecStart=%s\n[Install]\nWantedBy=multi-user.target\n' "unknown" "$start_line" > "/etc/systemd/system/$name.service"
```
Comment (mardy) [inline tests/lib/external/snapd-testing-tools/tools/tests.systemd:32]: Instead of hardcoding `"unknown"`, maybe we can use `$name` here?

**Review by mardy — COMMENTED** [review]
Nothing critical, just a couple of suggestions.

**Comment by sergiocazzolato** [comment]
@mardy hi, could you please review https://github.com/snapcore/snapd/pull/11995 instead, it is similar to this one but simple. I'll close this now, Thanks

**Pull request closed without merging**